### PR TITLE
WIP class refactor

### PR DESCRIPTION
**Pull request by bensquire**
This PR starts to add types into the shop class along with class properties. Hoping to get it merged in sooner rather that later to reduce the posibility of bugs.

Then I'll move onto one of the other classes.

**Note:** Quite a lot of indentation fixes in this PR.

### CODE REVIEW

File: _lib/core/shop.php [hunk 1]
```diff
@@ -161,6 +197,8 @@ public function shop()
 
     public function remove($product)
```
Comment (bensquire) [inline _lib/core/shop.php]: @adamjimenez What type is product?

File: _lib/core/shop.php [hunk 2]
```diff
@@ -246,15 +284,15 @@ public function get_basket()
         $this->vat = $this->subtotal * $this->vat_rate;
     }
 
-    public function add_to_basket($product, $quantity = 1, $buy_now = false, $variation_id = null, $extras_arr = '')
+    public function add_to_basket($product, int $quantity = 1, bool $buy_now = false, $variation_id = null, $extras_arr = '')
```
Comment (bensquire) [inline _lib/core/shop.php]: @adamjimenez What type is product, variation_id and extras_arr?

File: _lib/core/shop.php [hunk 3]
```diff
@@ -633,19 +671,19 @@ public function process_paypal()
     public function send_confirmation($oid)
```
Comment (bensquire) [inline _lib/core/shop.php]: @adamjimenez What type is oid?

File: _lib/core/shop.php [hunk 4]
```diff
@@ -775,14 +813,15 @@ public function complete_order($oid, $ref, $method)
         return false;
     }
 
-    public function failed_order($error, $oid, $ref, $method)
+    public function failed_order($error, $oid)
```
Comment (bensquire) [inline _lib/core/shop.php]: @adamjimenez What type is error?

File: _lib/core/shop.php (hunk 1 above)
Comment (adamjimenez) [reply _lib/core/shop.php]: int

File: _lib/core/shop.php (hunk 2 above)
Comment (adamjimenez) [reply _lib/core/shop.php]: product = int
variation_id = int
extras_arr = array

File: _lib/core/shop.php (hunk 3 above)
Comment (adamjimenez) [reply _lib/core/shop.php]: int

File: _lib/core/shop.php (hunk 4 above)
Comment (adamjimenez) [reply _lib/core/shop.php]: string

File: _lib/core/common.php [hunk 5]
```diff
@@ -548,7 +551,7 @@ function send_mail($opts = []): bool
     $is_html = ($opts['content'] !== strip_tags($opts['content']));
     
     if (getenv('SENDGRID_API_KEY')) {
-        $email = new SendGrid\Mail\Mail();
+        $email = new Mail();
```
Comment (adamjimenez) [inline _lib/core/common.php]: this will break things as I removed the namespace

File: _lib/core/common.php [hunk 6]
```diff
@@ -573,7 +576,7 @@ function send_mail($opts = []): bool
         
         return $response;
     } elseif (class_exists('PHPMailer\PHPMailer\PHPMailer')) {
-        $mail = new PHPMailer\PHPMailer\PHPMailer();
+        $mail = new PHPMailer();
```
Comment (adamjimenez) [inline _lib/core/common.php]: this will break things as I removed the namespace

File: _lib/core/common.php (hunk 5 above)
Comment (bensquire) [reply _lib/core/common.php]: The import statement is back at the top of the file, is that what you mean?

**Review by adamjimenez — COMMENTED** [review]
Yes, the imports should go as the classes are optional.

File: _lib/core/common.php (hunk 5 above)
Comment (bensquire) [reply _lib/core/common.php]: I think I've restored this?

File: _lib/core/common.php (hunk 6 above)
Comment (bensquire) [reply _lib/core/common.php]: I think I've restored this?

**Pull request merged**